kakounePlugins: use vim’s update.py script to generate plugins

Flakebi opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change
Similar to vim plugins, kakoune plugins are simple Git(Hub) repos that are easy to package. While this is tedious to do by hand, the update.py script makes it as convenient as 1. adding single line for the repo and 2. running a script and it even scales to half a thousand plugins (wohoo!).

#### General changes

To avoid duplicating the script, the kakoune update script imports vim’s script and only changes a few configuration variables and the output function. That needs a few updates in the vim script to make it generic over the editor, but it was already pretty generic in the first place.
The interesting commits in this regard is
0911a9e vimPlugins: make update.py generic over editor.

#### Kakoune specific changes

As it is simple with the script, I added a few kakoune plugins:
- alexherbo2/replace-mode.kak
- alexherbo2/sleuth.kak
- danr/kakoune-easymotion
- Delapouite/kakoune-registers
- enricozb/tabs.kak
- greenfork/active-window.kak
- JJK96/kakoune-rainbow
- kakoune-editor/kakoune-extra-filetypes
- occivink/kakoune-buffer-switcher

The plugins that were there previously are still available under the same name as before (aliased if the previous name did not match the git repository name) and seem to be still working (I didn’t try all of them but the generated code looks fine).

**Edit:** Some plugins are updated by this pr and I think their usage changed a bit. I have the following in my `kakrc` for powerline:
```js
hook global KakBegin .* %{
	powerline-start
	require-module powerline
	set global powerline_format 'bufname line_column position filetype mode_info'
	set global powerline_shorten_bufname short
	powerline-theme-tomorrow-night
}
```

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc kakoune plugin maintainers: @eraserhd, @nrdxp, @buffet
cc vim plugin maintainers: @jonringer, @softinio

Flakebi committed:
kakounePlugins: add buildKakounePlugin helper

Same as buildVimPlugin but for Kakoune

Flakebi committed:
kakounePlugins: use script for most plugins

All Kakoune plugins from before are still there and updated.

Added plugins:
- alexherbo2/connect.kak
- alexherbo2/replace-mode.kak
- alexherbo2/sleuth.kak
- danr/kakoune-easymotion
- Delapouite/kakoune-registers
- enricozb/tabs.kak
- greenfork/active-window.kak
- kakoune-editor/kakoune-extra-filetypes
- listentolist/kakoune-rainbow
- occivink/kakoune-buffer-switcher

Flakebi committed:
kakounePlugins: add aliases for old names

SuperSandro2000 commented:
Pinging @jcpetruzza because he did the last big overhaul.

eraserhd commented:
A few high-level thoughts:

I'm not fond of importing the vim script and monkey-patching it.  As much as I am normally opposed to copying code, this introduces a hidden dependency, and a surprising one.  If a Vim maintainer changes something trivial about that script, they would break Kakoune's plugins, which is "surprising" a la "principle of least surprise".  Copying it might be better.  Copying shared functionality into something intended to be a library, if there is one for python routines, would be a non-fragile way of deduplicating the functionality.

Boiling down to a repo list is nice, but it seems like every plugin would need to update `overrides.nix` anyway, at least for the license and probably for more meta.  That makes adding plugins a bit more confusing, and smells like "shotgun surgery".

Last, the default `installPhase` doesn't copy files to the kakoune autoload directory, so the plugins will not get loaded, unless they do so in `overrides.nix`.

Flakebi commented:
Thank you for the feedback @eraserhd!

> Moving shared functionality into something intended to be a library, if there is one for python routines, would be a non-fragile way of deduplicating the functionality.

Good idea, I’ll try to move the common parts into `maintainers/scripts/`.

The default `installPhase` copies the whole repository into the autoload directory (which is `rtpPath` in `build-kakoune-plugin.nix`). Most of the new plugins I added are not present in `overrides.nix`. The plugins in `overrides.nix` either have custom build or install steps or are not hosted in a GitHub repository.

> …at least for the license and probably for more meta.

Right, if a plugin is not added to `overrides.nix`, most metadata is missing (the script only sets `meta.homepage`). I hoped this would be fine as it is the same behaviour as for vim plugins.

Flakebi commented:
Rebased

SuperSandro2000 commented:
@Flakebi There are already a few merge conflicts.

SuperSandro2000 commented:
@Flakebi Again some merge conflicts. Can you give me a message on IRC when you fixed it so we can merge it before new merge conflicts are created.

Flakebi committed:
kakounePlugins: use recurseIntoAttrs

Flakebi commented:
Rebased and use `recurseIntoAttrs` for kakoune plugins so they show up in the search

SuperSandro2000 commented:
pinging some people to please take a look at this (again)
@eraserhd @wahjava @buffet

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108363` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
25 packages built:
</summary>
<ul>
<li>kakounePlugins.active-window-kak</li>
<li>kakounePlugins.auto-pairs-kak</li>
<li>kakounePlugins.case-kak</li>
<li>kakounePlugins.connect-kak</li>
<li>kakounePlugins.fzf-kak</li>
<li>kakounePlugins.kak-ansi</li>
<li>kakounePlugins.kak-plumb</li>
<li>kakounePlugins.kakoune-buffer-switcher</li>
<li>kakounePlugins.kakoune-buffers</li>
<li>kakounePlugins.kakoune-easymotion</li>
<li>kakounePlugins.kakoune-extra-filetypes</li>
<li>kakounePlugins.kakoune-rainbow</li>
<li>kakounePlugins.kakoune-registers</li>
<li>kakounePlugins.kakoune-state-save</li>
<li>kakounePlugins.kakoune-vertical-selection</li>
<li>kakounePlugins.openscad-kak</li>
<li>kakounePlugins.parinfer-rust</li>
<li>kakounePlugins.powerline-kak</li>
<li>kakounePlugins.prelude-kak</li>
<li>kakounePlugins.quickscope-kak</li>
<li>kakounePlugins.rep</li>
<li>kakounePlugins.replace-mode-kak</li>
<li>kakounePlugins.sleuth-kak</li>
<li>kakounePlugins.tabs-kak</li>
<li>spacevim</li>
</ul>
</details>

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108363` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
7 packages built:
</summary>
<ul>
<li>kakounePlugins.kak-ansi</li>
<li>kakounePlugins.kak-plumb</li>
<li>kakounePlugins.kakoune-state-save</li>
<li>kakounePlugins.parinfer-rust</li>
<li>kakounePlugins.quickscope-kak</li>
<li>kakounePlugins.rep</li>
<li>spacevim</li>
</ul>
</details>

jonringer commented on pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:
```diff
@@ -0,0 +1,35 @@
+    stdenv.mkDerivation (attrs // {
+      name = namePrefix + name;
+
+      inherit unpackPhase configurePhase buildPhase preInstall postInstall;
```
I don't think these are necessary, since they would be getting inherited from `attrs //`

jonringer replied:
also you may want to do `(lib.removeAttrs [ "namePrefix" ] attrs) //` and remove other values which aren't necessary for the underlying call to mkDerivation.

jonringer commented on pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:
```diff
@@ -0,0 +1,35 @@
+    });
+
+  buildKakounePluginFrom2Nix = attrs: buildKakounePlugin ({
+    buildPhase = ":";
+    configurePhase =":";
```
```suggestion
    dontBuild = true;
    dontConfigure = true;
```

Flakebi replied:
I tried setting it to either `true` or `false`, but none of the works. Some plugins (e.g. [connect-kak](https://github.com/alexherbo2/connect.kak)) have a Makefile in the repository which then gets executed and fails.
Setting it to `"true"` seems to work. I left it unchanged at `":"`, which is consistent with the vim plugins.

jonringer reviewed (commented):
otherwise LGTM

Flakebi commented:
Rebased

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108363` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
25 packages built:
</summary>
<ul>
<li>kakounePlugins.active-window-kak</li>
<li>kakounePlugins.auto-pairs-kak</li>
<li>kakounePlugins.case-kak</li>
<li>kakounePlugins.connect-kak</li>
<li>kakounePlugins.fzf-kak</li>
<li>kakounePlugins.kak-ansi</li>
<li>kakounePlugins.kak-plumb</li>
<li>kakounePlugins.kakoune-buffer-switcher</li>
<li>kakounePlugins.kakoune-buffers</li>
<li>kakounePlugins.kakoune-easymotion</li>
<li>kakounePlugins.kakoune-extra-filetypes</li>
<li>kakounePlugins.kakoune-rainbow</li>
<li>kakounePlugins.kakoune-registers</li>
<li>kakounePlugins.kakoune-state-save</li>
<li>kakounePlugins.kakoune-vertical-selection</li>
<li>kakounePlugins.openscad-kak</li>
<li>kakounePlugins.parinfer-rust</li>
<li>kakounePlugins.powerline-kak</li>
<li>kakounePlugins.prelude-kak</li>
<li>kakounePlugins.quickscope-kak</li>
<li>kakounePlugins.rep</li>
<li>kakounePlugins.replace-mode-kak</li>
<li>kakounePlugins.sleuth-kak</li>
<li>kakounePlugins.tabs-kak</li>
<li>spacevim</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
kakounePlugins.active-window-kak:
<details>


Package does not have a maintainer. Consider adding yourself?

Near pkgs/applications/editors/kakoune/plugins/generated.nix:15:5:

```
   |
15 |     meta.homepage = "https://github.com/greenfork/active-window.kak/";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
Package is missing a license

Near pkgs/applications/editors/kakoune/plugins/generated.nix:15:5:

```
   |
15 |     meta.homepage = "https://github.com/greenfork/active-window.kak/";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/license-missing.md
`configurePhase` should probably contain `runHook preConfigure` and `runHook postConfigure`.

Near pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:31:5:

```
   |
31 |     configurePhase = ":";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:30:5:

```
   |
30 |     buildPhase = ":";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
</details>
kakounePlugins.auto-pairs-kak:
<details>


Package does not have a maintainer. Consider adding yourself?

Near pkgs/applications/editors/kakoune/plugins/generated.nix:27:5:

```
   |
27 |     meta.homepage = "https://github.com/alexherbo2/auto-pairs.kak/";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
Package is missing a license

Near pkgs/applications/editors/kakoune/plugins/generated.nix:27:5:

```
   |
27 |     meta.homepage = "https://github.com/alexherbo2/auto-pairs.kak/";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/license-missing.md
`configurePhase` should probably contain `runHook preConfigure` and `runHook postConfigure`.

Near pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:31:5:

```
   |
31 |     configurePhase = ":";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:30:5:

```
   |
30 |     buildPhase = ":";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
</details>
kakounePlugins.case-kak:
<details>


Package does not have a maintainer. Consider adding yourself?

Near pkgs/applications/editors/kakoune/plugins/overrides.nix:19:5:

```
   |
19 |     meta.homepage = "https://gitlab.com/FlyingWombat/case.kak";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
Package is missing a license

Near pkgs/applications/editors/kakoune/plugins/overrides.nix:19:5:

```
   |
19 |     meta.homepage = "https://gitlab.com/FlyingWombat/case.kak";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/license-missing.md
`configurePhase` should probably contain `runHook preConfigure` and `runHook postConfigure`.

Near pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:31:5:

```
   |
31 |     configurePhase = ":";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:30:5:

```
   |
30 |     buildPhase = ":";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
</details>
kakounePlugins.connect-kak:
<details>


Package does not have a maintainer. Consider adding yourself?

Near pkgs/applications/editors/kakoune/plugins/generated.nix:39:5:

```
   |
39 |     meta.homepage = "https://github.com/alexherbo2/connect.kak/";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
Package is missing a license

Near pkgs/applications/editors/kakoune/plugins/generated.nix:39:5:

```
   |
39 |     meta.homepage = "https://github.com/alexherbo2/connect.kak/";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/license-missing.md
`configurePhase` should probably contain `runHook preConfigure` and `runHook postConfigure`.

Near pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:31:5:

```
   |
31 |     configurePhase = ":";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:30:5:

```
   |
30 |     buildPhase = ":";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
</details>
kakounePlugins.kak-ansi:
<details>


`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/applications/editors/kakoune/plugins/overrides.nix:46:5:

```
   |
46 |     installPhase = ''
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
</details>
kakounePlugins.kakoune-buffers:
<details>


Package does not have a maintainer. Consider adding yourself?

Near pkgs/applications/editors/kakoune/plugins/generated.nix:75:5:

```
   |
75 |     meta.homepage = "https://github.com/Delapouite/kakoune-buffers/";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
Package is missing a license

Near pkgs/applications/editors/kakoune/plugins/generated.nix:75:5:

```
   |
75 |     meta.homepage = "https://github.com/Delapouite/kakoune-buffers/";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/license-missing.md
`configurePhase` should probably contain `runHook preConfigure` and `runHook postConfigure`.

Near pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:31:5:

```
   |
31 |     configurePhase = ":";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:30:5:

```
   |
30 |     buildPhase = ":";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
</details>
kakounePlugins.kakoune-buffer-switcher:
<details>


Package does not have a maintainer. Consider adding yourself?

Near pkgs/applications/editors/kakoune/plugins/generated.nix:63:5:

```
   |
63 |     meta.homepage = "https://github.com/occivink/kakoune-buffer-switcher/";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
Package is missing a license

Near pkgs/applications/editors/kakoune/plugins/generated.nix:63:5:

```
   |
63 |     meta.homepage = "https://github.com/occivink/kakoune-buffer-switcher/";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/license-missing.md
`configurePhase` should probably contain `runHook preConfigure` and `runHook postConfigure`.

Near pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:31:5:

```
   |
31 |     configurePhase = ":";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:30:5:

```
   |
30 |     buildPhase = ":";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
</details>
kakounePlugins.kakoune-easymotion:
<details>


Package does not have a maintainer. Consider adding yourself?

Near pkgs/applications/editors/kakoune/plugins/generated.nix:87:5:

```
   |
87 |     meta.homepage = "https://github.com/danr/kakoune-easymotion/";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
Package is missing a license

Near pkgs/applications/editors/kakoune/plugins/generated.nix:87:5:

```
   |
87 |     meta.homepage = "https://github.com/danr/kakoune-easymotion/";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/license-missing.md
`configurePhase` should probably contain `runHook preConfigure` and `runHook postConfigure`.

Near pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:31:5:

```
   |
31 |     configurePhase = ":";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:30:5:

```
   |
30 |     buildPhase = ":";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
</details>
kakounePlugins.kakoune-extra-filetypes:
<details>


Package does not have a maintainer. Consider adding yourself?

Near pkgs/applications/editors/kakoune/plugins/generated.nix:99:5:

```
   |
99 |     meta.homepage = "https://github.com/kakoune-editor/kakoune-extra-filetypes/";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
Package is missing a license

Near pkgs/applications/editors/kakoune/plugins/generated.nix:99:5:

```
   |
99 |     meta.homepage = "https://github.com/kakoune-editor/kakoune-extra-filetypes/";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/license-missing.md
`configurePhase` should probably contain `runHook preConfigure` and `runHook postConfigure`.

Near pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:31:5:

```
   |
31 |     configurePhase = ":";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:30:5:

```
   |
30 |     buildPhase = ":";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
</details>
kakounePlugins.kakoune-registers:
<details>


Package does not have a maintainer. Consider adding yourself?

Near pkgs/applications/editors/kakoune/plugins/generated.nix:123:5:

```
    |
123 |     meta.homepage = "https://github.com/Delapouite/kakoune-registers/";
    |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
Package is missing a license

Near pkgs/applications/editors/kakoune/plugins/generated.nix:123:5:

```
    |
123 |     meta.homepage = "https://github.com/Delapouite/kakoune-registers/";
    |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/license-missing.md
`configurePhase` should probably contain `runHook preConfigure` and `runHook postConfigure`.

Near pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:31:5:

```
   |
31 |     configurePhase = ":";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:30:5:

```
   |
30 |     buildPhase = ":";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
</details>
kakounePlugins.kakoune-state-save:
<details>


`configurePhase` should probably contain `runHook preConfigure` and `runHook postConfigure`.

Near pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:31:5:

```
   |
31 |     configurePhase = ":";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:30:5:

```
   |
30 |     buildPhase = ":";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
</details>
kakounePlugins.kakoune-vertical-selection:
<details>


Package does not have a maintainer. Consider adding yourself?

Near pkgs/applications/editors/kakoune/plugins/generated.nix:135:5:

```
    |
135 |     meta.homepage = "https://github.com/occivink/kakoune-vertical-selection/";
    |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
Package is missing a license

Near pkgs/applications/editors/kakoune/plugins/generated.nix:135:5:

```
    |
135 |     meta.homepage = "https://github.com/occivink/kakoune-vertical-selection/";
    |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/license-missing.md
`configurePhase` should probably contain `runHook preConfigure` and `runHook postConfigure`.

Near pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:31:5:

```
   |
31 |     configurePhase = ":";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:30:5:

```
   |
30 |     buildPhase = ":";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
</details>
kakounePlugins.kak-plumb:
<details>


`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/applications/editors/kakoune/plugins/overrides.nix:77:5:

```
   |
77 |     installPhase = ''
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
</details>
kakounePlugins.openscad-kak:
<details>


Package does not have a maintainer. Consider adding yourself?

Near pkgs/applications/editors/kakoune/plugins/generated.nix:147:5:

```
    |
147 |     meta.homepage = "https://github.com/mayjs/openscad.kak/";
    |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
Package is missing a license

Near pkgs/applications/editors/kakoune/plugins/generated.nix:147:5:

```
    |
147 |     meta.homepage = "https://github.com/mayjs/openscad.kak/";
    |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/license-missing.md
`configurePhase` should probably contain `runHook preConfigure` and `runHook postConfigure`.

Near pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:31:5:

```
   |
31 |     configurePhase = ":";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:30:5:

```
   |
30 |     buildPhase = ":";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
</details>
kakounePlugins.prelude-kak:
<details>


Package does not have a maintainer. Consider adding yourself?

Near pkgs/applications/editors/kakoune/plugins/generated.nix:171:5:

```
    |
171 |     meta.homepage = "https://github.com/alexherbo2/prelude.kak/";
    |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
Package is missing a license

Near pkgs/applications/editors/kakoune/plugins/generated.nix:171:5:

```
    |
171 |     meta.homepage = "https://github.com/alexherbo2/prelude.kak/";
    |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/license-missing.md
`configurePhase` should probably contain `runHook preConfigure` and `runHook postConfigure`.

Near pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:31:5:

```
   |
31 |     configurePhase = ":";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:30:5:

```
   |
30 |     buildPhase = ":";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
</details>
kakounePlugins.quickscope-kak:
<details>


`configurePhase` should probably contain `runHook preConfigure` and `runHook postConfigure`.

Near pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:31:5:

```
   |
31 |     configurePhase = ":";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:30:5:

```
   |
30 |     buildPhase = ":";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
</details>
kakounePlugins.replace-mode-kak:
<details>


Package does not have a maintainer. Consider adding yourself?

Near pkgs/applications/editors/kakoune/plugins/generated.nix:183:5:

```
    |
183 |     meta.homepage = "https://github.com/alexherbo2/replace-mode.kak/";
    |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
Package is missing a license

Near pkgs/applications/editors/kakoune/plugins/generated.nix:183:5:

```
    |
183 |     meta.homepage = "https://github.com/alexherbo2/replace-mode.kak/";
    |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/license-missing.md
`configurePhase` should probably contain `runHook preConfigure` and `runHook postConfigure`.

Near pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:31:5:

```
   |
31 |     configurePhase = ":";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:30:5:

```
   |
30 |     buildPhase = ":";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
</details>
kakounePlugins.sleuth-kak:
<details>


Package does not have a maintainer. Consider adding yourself?

Near pkgs/applications/editors/kakoune/plugins/generated.nix:195:5:

```
    |
195 |     meta.homepage = "https://github.com/alexherbo2/sleuth.kak/";
    |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
Package is missing a license

Near pkgs/applications/editors/kakoune/plugins/generated.nix:195:5:

```
    |
195 |     meta.homepage = "https://github.com/alexherbo2/sleuth.kak/";
    |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/license-missing.md
`configurePhase` should probably contain `runHook preConfigure` and `runHook postConfigure`.

Near pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:31:5:

```
   |
31 |     configurePhase = ":";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:30:5:

```
   |
30 |     buildPhase = ":";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
</details>
kakounePlugins.tabs-kak:
<details>


Package does not have a maintainer. Consider adding yourself?

Near pkgs/applications/editors/kakoune/plugins/generated.nix:207:5:

```
    |
207 |     meta.homepage = "https://github.com/enricozb/tabs.kak/";
    |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md
Package is missing a license

Near pkgs/applications/editors/kakoune/plugins/generated.nix:207:5:

```
    |
207 |     meta.homepage = "https://github.com/enricozb/tabs.kak/";
    |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/license-missing.md
`configurePhase` should probably contain `runHook preConfigure` and `runHook postConfigure`.

Near pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:31:5:

```
   |
31 |     configurePhase = ":";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/applications/editors/kakoune/plugins/build-kakoune-plugin.nix:30:5:

```
   |
30 |     buildPhase = ":";
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
</details>
spacevim:
<details>


`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/applications/editors/spacevim/default.nix:27:3:

```
   |
27 |   buildPhase = ''
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/applications/editors/spacevim/default.nix:34:3:

```
   |
34 |   installPhase = ''
   |   ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md
Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/applications/editors/spacevim/default.nix:32:15:

```
   |
32 |   patches = [ ./helptags.patch ];
   |               ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md
</details>
</details>

nrdxp reviewed (commented):
Looks good, only possible nitpick would be to execute a rename command as a separate commit, so that the semantic diffs are preserved in a following commit. Add we are not adding/removing huge chunks.

Flakebi committed:
vimPlugins: make update.py generic over editor

Move the script to maintainers/scripts/pluginupdate.py.
Importing it from the vim and kakoune update scripts
is done in the commit afterwards to cleanup the diff.

Flakebi committed:
vimPlugins: add vim and kakoune update script

Flakebi committed:
kakounePlugins: update

Flakebi commented:
Ah, I didn’t notice the diff of the update script is that unreadable. It’s in a separate commit now and much easier to read, thanks.

nrdxp reviewed (approved):
I believe this is ready for merging. I would do it myself if I had access. Good work!

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-already-reviewed/2617/354

Flakebi commented:
Rebased

AndersonTorres commented:
Many thanks!

rmcgibbo commented:
@Flakebi / @AndersonTorres: Looks like one of these might have the wrong github revision?

```
$ nix build -f . kakounePlugins.powerline-kak
builder for '/nix/store/jzr6672x19bpzwapvncry8h1rsw9g8k9-source.drv' failed with exit code 1; last 7 log lines:
  
  trying https://github.com/jdugan6240/powerline.kak/archive/322a760daa099d519ff50d14c29b27f3e2af00d1.tar.gz
    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                   Dload  Upload   Total   Spent    Left  Speed
    0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  curl: (22) The requested URL returned error: 404 
  error: cannot download source from any mirror
cannot build derivation '/nix/store/ab4z3f4rw1vh0gcrn3wf6ngw9407wpn6-kakplugin-powerline-kak-2021-02-15.drv': 1 dependencies couldn't be built
[0 built (1 failed)]
error: build of '/nix/store/ab4z3f4rw1vh0gcrn3wf6ngw9407wpn6-kakplugin-powerline-kak-2021-02-15.drv' failed
```

nrdxp commented:
Actually, `jdugan6240/powerline.kak` doesn't exist anymore. He was originally supposed to take over maintainership from `andreyorst/powerline.kak`, but it appears it has actually shifted back the original location recently.

Just need to change [this](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/editors/kakoune/plugins/kakoune-plugin-names#L12) to `andreyorst/powerline.kak`, or if another maintainer has been chosen that I'm not aware of, to the new maintainer's location.

Flakebi commented:
Seems like it’s maintained by andreyorst again! https://github.com/andreyorst/powerline.kak/pull/18
There were also some other repository moves, I opened #116485.

colemickens commented:
Am I missing something or did this drop the ability to set an auth token to use with GitHub? This is sort of an enormous pain to use without it... I can re-patch it if need be.

colemickens commented:
never mind, someone has informed me that I indeed did miss something (the rest of the refactor, clearly). ignore me!
